### PR TITLE
feat(platform): support cilium overlay underlay network modes

### PR DESCRIPTION
**Pull request by Huimintai**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/platform/provider/baremetal/manifests/cilium/router.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    operator: NotIn
+                    values:
+                      - hollow
+      containers:
+        - args:
+            - -iface=$(HOST_IP)
+            - -node-name=$(NODE_NAME)
+            - -cluster-ip-range={{ .ClusterCIDR }}
+            - -network-mode=bgp
+            - -bgp-peer-auth=
```
Comment (huxiaoliang) [inline pkg/platform/provider/baremetal/manifests/cilium/router.yaml:82]: remove `bgp-peer-auth` if no value specified

File: pkg/platform/provider/baremetal/manifests/cilium/eni-ipamd.yaml [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+      labels:
+        k8s-app: tke-eni-ipamd
+        qcloud-app: tke-eni-ipamd
+    spec:
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - preference:
+                matchExpressions:
+                  - key: node-role.kubernetes.io/master
```
Comment (leoryu) [inline pkg/platform/provider/baremetal/manifests/cilium/eni-ipamd.yaml:145]: should handle 1.20+ k8s, maybe just has `node-role.kubernetes.io/control-plane`.

Comment (leoryu) [reply pkg/platform/provider/baremetal/manifests/cilium/eni-ipamd.yaml:145]: add 
        - matchExpressions:
              - key: node-role.kubernetes.io/control-plane
                operator: Exists

Comment (leoryu) [reply pkg/platform/provider/baremetal/manifests/cilium/eni-ipamd.yaml:145]: ref https://github.com/tkestack/tke/pull/1293/files

**Pull request merged**